plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -392,30 +392,33 @@ public IcebergTableHandle getTableHandle(
     private static long getSnapshotIdFromVersion(Table table, ConnectorTableVersion version)
     {
         io.trino.spi.type.Type versionType = version.getVersionType();
-        switch (version.getPointerType()) {
-            case TEMPORAL:
-                long epochMillis;
-                if (versionType instanceof TimestampWithTimeZoneType) {
-                    epochMillis = ((TimestampWithTimeZoneType) versionType).isShort()
-                            ? unpackMillisUtc((long) version.getVersion())
-                            : ((LongTimestampWithTimeZone) version.getVersion()).getEpochMillis();
-                }
-                else {
-                    throw new TrinoException(NOT_SUPPORTED, "Unsupported type for temporal table version: " + versionType.getDisplayName());
-                }
-                return getSnapshotIdAsOfTime(table, epochMillis);
+        return switch (version.getPointerType()) {
+            case TEMPORAL -> getTemporalSnapshotIdFromVersion(table, version, versionType);
+            case TARGET_ID -> getTargetSnapshotIdFromVersion(table, version, versionType);
```

Comment:
Optional: Could you add a preparatory commit to extract methods?

Reply:
added: [Extract getting snapshotId from Iceberg metadata by types](https://github.com/trinodb/trino/pull/15395/commits/5f2ccea44af812f25f079b4e84c310cc853e4d9e)